bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/handler/VerisureBridgeHandler.java

```diff
@@ -171,15 +172,17 @@ public void dispose() {
         }
     }
 
-    public boolean registerObjectStatusListener(DeviceStatusListener deviceStatusListener) {
+    public <T extends VerisureThing> boolean registerObjectStatusListener(
+            DeviceStatusListener<T> deviceStatusListener) {
         if (session != null) {
             logger.debug("registerObjectStatusListener for listener {}", deviceStatusListener);
             return session.registerDeviceStatusListener(deviceStatusListener);
         }
```

Comment:
@jannegpriv Well after messing around I found that what you have so far works, and trying to add more generics just makes things worse. So I think the generics are good enough for now. 🙂 
The bigger issue I saw were all the null check warnings that we would have to resolve. This is not as bad as dealing with generics but still requires some minor workarounds to deal with.

A majority of the complaints can be dealt with by creating a local variable for class fields like so:
```suggestion
        VerisureSession session = this.session;
        if (session != null) {
            logger.debug("registerObjectStatusListener for listener {}", deviceStatusListener);
            return session.registerDeviceStatusListener(deviceStatusListener);
        }
```

Reply:
> @jannegpriv Well after messing around I found that what you have so far works, and trying to add more generics just makes things worse. So I think the generics are good enough for now. 🙂

gr8! :-)

> A majority of the complaints can be dealt with by creating a local variable for class fields like soot:

I will try to fix them, but those warnings are not correct since we have null checks, why are they flagged as compiler warnings?

Reply:
I've now fixed those warnings.

I've got another question, if I remove the `@NonNullByDefault` declarations on inner classes I get a report file from static code analyse stating that those are needed. If I keep them I get a compiler warning stating:

```
Nullness default is redundant with a default specified for the enclosing type VerisureSmartPlugs
```

What is the correct way of doing it?

Reply:
Damned if you do, damned if you don't kinda situation there. I think the compiler warnings trump static code analysis warnings so remove them from the inner classes.

Reply:
OK, will do!

Reply:
I think I'll keep the `@NonNullByDefault` on inner classes since reoving causes both warnings and a report file being generated, so the build log looks much cleaner with them present.

Reply:
@cpmeister What's left to be done?

Reply:
>  removing them causes both warnings and a report file being generated

In the next SAT version it will only check for `@NonNullByDefault` on outer classes which should fix a lot of warnings and needless redundancy, see https://github.com/openhab/static-code-analysis/pull/383.